qrm/scan_context.go

```diff
@@ -222,19 +222,23 @@ func (s *scanContext) rowElem(index int) interface{} {
 	return value
 }
 
-func (s *scanContext) rowElemValuePtr(index int) reflect.Value {
+func (s *scanContext) rowElemValuePtr(index int) *reflect.Value {
 	rowElem := s.rowElem(index)
 	rowElemValue := reflect.ValueOf(rowElem)
 
 	if rowElemValue.Kind() == reflect.Ptr {
-		return rowElemValue
+		return &rowElemValue
 	}
 
 	if rowElemValue.CanAddr() {
-		return rowElemValue.Addr()
+		addr := rowElemValue.Addr()
+		return &addr
 	}
 
-	newElem := reflect.New(rowElemValue.Type())
-	newElem.Elem().Set(rowElemValue)
-	return newElem
+	if rowElemValue.IsValid() {
+		newElem := reflect.New(rowElemValue.Type())
+		newElem.Elem().Set(rowElemValue)
+		return &newElem
+	}
+	return nil
```

Comment:
I could returns `reflect.Value{}` instead of `nil` and not change the signature of this function. We need to verify for `IsValid` instead of `!= nil` in the calling function.

Reply:
Agreed, I think that would be a better approach.

Reply:
Agree, I think that would be a bit cleaner approach.

Reply:
Updated

Reply:
LGTM.